### PR TITLE
[Merged by Bors] - feat: positivity of `star x * x`

### PR DESCRIPTION
**Pull request by eric-wieser**
It is positive when `IsRegular x` (which requires weaker constraints on `R` than `IsUnit x` or `x ≠ 0`)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Star/Order.lean [hunk 1]
```diff
@@ -233,6 +234,16 @@ lemma IsSelfAdjoint.mono {x y : R} (h : x ≤ y) (hx : IsSelfAdjoint x) : IsSelf
 lemma IsSelfAdjoint.of_nonneg {x : R} (hx : 0 ≤ x) : IsSelfAdjoint x :=
   (isSelfAdjoint_zero R).mono hx
 
+theorem star_mul_self_pos [Nontrivial R] {x : R} (hx : IsRegular x) : 0 < star x * x := by
+  rw [lt_iff_le_and_ne]
+  refine ⟨star_mul_self_nonneg _, ?_⟩
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Star/Order.lean:239]:
```suggestion
  rw [(star_mul_self_nonneg _).lt_iff_ne]
```

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm

**Comment by j-loreaux** [comment]
This is weird to me because `IsRegular` seems *so* much stronger than `x ≠ 0`. In fact, `(hx : 0 ≤ x) (hx' : IsRegular x)` is basically the definition of *strictly positive* in a C⋆-algebra (although there are other equivalent ones). Nevertheless, I think there's nothing wrong with having these.

bors merge

**Comment by eric-wieser** [comment]
> This is weird to me because `IsRegular` seems _so_ much stronger than `x ≠ 0`.

I think I stated things unclearly; what I meant is that `[Field R] (x : R) (hx : x ≠ 0)` is a much stronger requirement than `[Ring R] (x : R) (hx : IsRegular R)`.

**Pull request closed without merging**